dotnetv3/cross-service/DynamoDbItemTracker/ItemTrackerTests/WorkItemServiceTests.cs

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /// <summary>
+    /// Validate getting all items.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(3)]
+    [Trait("Category", "Unit")]
+    public async Task VerifyGetItems_ShouldReturnItems()
+    {
+        var mockContext = new Mock<IDynamoDBContext>();
+        var mockScan = new Mock<AsyncSearch<WorkItem>>();
+        var response = new List<WorkItem> { new WorkItem(), new WorkItem() };
+
+        // Only return the mock collection if there is no filter.
+
+        mockScan.Setup(ms =>
+            ms.GetRemainingAsync(CancellationToken.None).Result).Returns(response);
+
+        mockContext.Setup(mc =>
+            mc.FromScanAsync<WorkItem>(
+                It.Is<ScanOperationConfig>(r => r.Filter.ToConditions().Count == 0),
+                It.IsAny<DynamoDBOperationConfig>())).Returns(mockScan.Object);
+
+        var service = new WorkItemService(mockContext.Object);
+        var items = await service.GetAllItems();
+        Assert.NotEmpty(items);
+    }
+
+    /// <summary>
+    /// Validate getting items by archive status.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(4)]
+    [Trait("Category", "Unit")]
+    public async Task VerifyGetItemsByArchiveState_ShouldReturnItems()
+    {
+        var mockContext = new Mock<IDynamoDBContext>();
+        var mockScan = new Mock<AsyncSearch<WorkItem>>();
+        var response = new List<WorkItem> { new WorkItem(), new WorkItem() };
+
+        // Only return the mock collection if there is no filter.
+
+        mockScan.Setup(ms =>
+            ms.GetRemainingAsync(CancellationToken.None).Result).Returns(response);
+
+        mockContext.Setup(mc =>
+            mc.FromScanAsync<WorkItem>(
+                It.Is<ScanOperationConfig>(r => r.Filter.ToConditions().ContainsKey("archived")),
+                It.IsAny<DynamoDBOperationConfig>())).Returns(mockScan.Object);
+
+        var service = new WorkItemService(mockContext.Object);
+        var items = await service.GetItemsByArchiveState(false);
+        Assert.Equal(2, items.Count);
+    }
+
+    /// <summary>
+    /// Validate getting an item by ID.
+    /// </summary>
+    /// <returns>Async task.</returns>
+    [Fact]
+    [Order(5)]
+    [Trait("Category", "Unit")]
+    public async Task VerifyGetItemById_ShouldReturnItem()
+    {
+        var testId = "testId";
+        var testWorkItem = new WorkItem() { Id = testId };
+        var mockContext = new Mock<IDynamoDBContext>();
+        mockContext.Setup(mc =>
+            mc.LoadAsync<WorkItem>(
+                It.Is<string>(id => id == testId), CancellationToken.None)).ReturnsAsync(testWorkItem);
+
+        var service = new WorkItemService(mockContext.Object);
+        var item = await service.GetItem(testId);
+        Assert.Equal(testId, item.iditem);
+    }
+
+    /// <summary>
+    /// Validate creating an item executes the save and load methods and sets an Id.
```

Comment:
Validate creating --> Validate that creating

Reply:
Fixed.